dev/release/post-07-csharp.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+# -*- indent-tabs-mode: nil; sh-indentation: 2; sh-basic-offset: 2 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+set -e
+set -u
+set -o pipefail
+
+main() {
+  if [ "$#" -ne 1 ]; then
+    echo "Usage: $0 <version>"
+    exit
+  fi
+
+  local -r version="$1"
+
+  if [ -z "${NUGET_API_KEY}" ]; then
+    echo "NUGET_API_KEY is empty"
+    exit 1
+  fi
+
+  local base_names=()
+  base_names+=(Apache.Arrow.Adbc.${version})
+  base_names+=(Apache.Arrow.Adbc.Client.${version})
+  base_names+=(Apache.Arrow.Adbc.Drivers.BigQuery.${version})
+  base_names+=(Apache.Arrow.Adbc.Drivers.FlightSql.${version})
+  base_names+=(Apache.Arrow.Adbc.Drivers.Interop.Snowflake.${version})
+  for base_name in ${base_names[@]}; do
```

Comment:
In general, we should use `"${array[@]}"` form:

```suggestion
  for base_name in "${base_names[@]}"; do
```

Reply:
Applied suggestion